daiquiri/query/static/query/js/services/plot.js

```diff
@@ -1,208 +1,569 @@
-        }
+    /* x and y will be always defined (unless query results do not have data */
+    var x_params = angular.extend({}, service.params, {
+      column: service.values.x
+    });
+    var y_params = angular.extend({}, service.params, {
+      column: service.values.y
+    });
 
-        if (service.values.y) {
-            var y_column = $filter('filter')(service.columns, {name: service.values.y}, true)[0];
+    var res_promises = [resources.rows.paginate(x_params).$promise,
+                        resources.rows.paginate(y_params).$promise];
 
-            if (['int', 'long', 'float', 'double'].indexOf(y_column.datatype) > -1) {
-                service.errors.y = null;
+    /* add y2 and y3 only if there are selected on the html form*/
+    if (service.values.y2){
+      var y2_params = angular.extend({}, service.params, {
+        column: service.values.y2
+      });
+      res_promises.push(resources.rows.paginate(y2_params).$promise);
+    }
 
-                service.labels.y = y_column.name;
-                if (y_column.unit && y_column.unit.length > 0) {
-                    service.labels.y += ' [' + y_column.unit + ']';
-                }
-            } else {
-                service.errors.y = y_column.datatype;
-            }
-        } else {
-            service.errors.y = 'empty';
-        }
+    if (service.values.y3){
+      var y3_params = angular.extend({}, service.params, {
+        column: service.values.y3
+      });
+      res_promises.push(resources.rows.paginate(y3_params).$promise)
+    }
 
-        if (service.errors.x === null && service.errors.y === null) {
-            service.fetch().then(function() {
-                service.draw();
-                service.ready = true;
-            });
-        } else {
-            service.ready = true;
+    /* fill service.source with the results from the query*/
+    return $q.all(res_promises).then(function(results) {
+      service.source = new Bokeh.ColumnDataSource({
+        data: {
+          x: results[0].results,
+          y: results[1].results
         }
-    };
+      });
+      if (service.values.y2){service.source.data.y2 = results[2].results;}
+      if (service.values.y3){service.source.data.y3 = results[3].results;}
+    });
+  }
+
+  /* create a basic figure template */
+  create_figure = function(x_range, y_range){
+
+    var figure_options = {
+      width: 840,
+      height: 500,
+      x_range: x_range,
+      y_range: y_range,
+      plot_width: $('.col-md-9').width(),
+      tools: 'wheel_zoom,box_zoom,reset,save',
+      x_axis_label: service.labels.x,
+      y_axis_label: service.labels.y,
+      background_fill_color: '#f5f5f5'
+    }
+
+    if (service.webgl) {
+      figure_options['output_backend'] = 'webgl';
+    }
+
+    var figure = new Bokeh.Plotting.figure(figure_options);
+
+    figure.toolbar.active_scroll = figure.toolbar.wheel_zoom;
+    figure.outline_line_color = '#dddddd';
+    figure.toolbar.logo = null;
```

Comment:
I would have no problem to make advertisement for Bokeh. To my opinion the logo can stay.